code/modules/mob/mob.dm

```diff
@@ -42,108 +42,100 @@ var/next_mob_id = 0
 	for(var/datum/gas/trace_gas in environment.trace_gases)
 		t+= "<span class='notice'>[trace_gas.type]: [trace_gas.moles] \n</span>"
 
-	usr.show_message(t, 1)
+	usr << t
 
-/mob/proc/show_message(msg, type, alt, alt_type)//Message, type of message (1 or 2), alternative message, alt message type (1 or 2)
+/mob/proc/show_message(msg, type, alt_msg, alt_type)//Message, type of message (1 or 2), alternative message, alt message type (1 or 2)
 
-	if(!client)	return
+	if(!client)
+		return
 
 	msg = copytext(msg, 1, MAX_MESSAGE_LEN)
 
-	if (type)
-		if(type & 1 && (disabilities & BLIND || paralysis) )//Vision related
-			if (!( alt ))
+	if(type)
+		if(type & 1 && eye_blind )//Vision related
+			if(!alt_msg)
 				return
 			else
-				msg = alt
+				msg = alt_msg
 				type = alt_type
-		if (type & 2 && ear_deaf)//Hearing related
-			if (!( alt ))
+
+		if(type & 2 && ear_deaf)//Hearing related
+			if(!alt_msg)
 				return
 			else
-				msg = alt
+				msg = alt_msg
 				type = alt_type
-				if ((type & 1 && disabilities & BLIND))
+				if(type & 1 && eye_blind)
 					return
-	// Added voice muffling for Issue 41.
-	if(stat == UNCONSCIOUS || sleeping > 0)
-		src << "<I>... You can almost hear someone talking ...</I>"
+	// voice muffling
+	if(stat == UNCONSCIOUS)
+		if(type & 2) //audio
+			src << "<I>... You can almost hear something ...</I>"
```

Comment:
I changed the wording a bit because this does not cover only say speech. The check is here so you don't get that message when unconscious and a blob attacks you or you're hit by a floral somatoray beam for example.